lib/teiserver_web/controllers/admin/user_controller.ex

```diff
@@ -1104,4 +1152,24 @@ defmodule TeiserverWeb.Admin.UserController do
         |> redirect(to: ~p"/teiserver/admin/user")
     end
   end
+
+
+  @spec delete_user(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def delete_user(conn, %{"id" => id}) do
+    user = Account.get_user_by_id(id)
+
+    case Teiserver.Account.UserLib.has_access(user, conn) do
```

Comment:
:+1: for the access control here. I'd recommend using that approach or the one I suggested in both functions.